### PR TITLE
Validations, testing models and  n+1 problems

### PR DESCRIPTION
**Pull request by estebanmual**
In this milestone I:
  - Added validations for all the models
  - Added unit specs for all the models' methods and validations.
  - Added flash messages in the create actions in all your controllers.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+    post = Post.new(Title: 'test', Text: 'testing post', user_id: 6)
+    post.save
+    expect(post).to be_valid
+  end
+
+  it 'is not valid' do
+    post = Post.new(Text: 'testing post', user_id: 6)
+    post.save
+    expect(post).to_not be_valid
+  end
```
Comment (ahmetbozaci) [inline spec/models/post_spec.rb]: - [x] Please also add test for these validations:

* Title must not exceed 250 characters.
* CommentsCounter must be an integer greater than or equal zero.
* LikesCounter must be an integer greater than or equal zero.

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


 @estebanmual  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Nice to have 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
- Nice to add validations on the user and post model according to requirements.✔️
- Flash messages work well when creating posts, like, and comments.✔️
___


### ♻️ Request changes ♻️

- [x] _Check the comments under the review._
- [x] You should fix the n+1 problem in the post controller. [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec_n%2B1.md#:~:text=Make%20sure%20that%20the%20N%2B1%20problem%20is%20solved%20when%20fetching%20all%20posts%20and%20their%20comments%20for%20a%20user.%20This%20is%20the%20corresponding%20wireframe)
`@posts = @user.posts.includes(:comments)` instead of `@posts = @user.posts`
image 
- [x] Please fix the errors in the tests files.
image
- [x] You should also add tests for custom methods.(`update_post_counter, recent_comment, recent_posts, update_like_counter, update_comment_count`)  [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec_n%2B1.md#:~:text=all%20of%20your-,models%27%20methods,-and%20validations.)

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- Your commit messages should always start with capital letters and they should be an imperative tense. I see some of them are not like that. Instead of writing `Added style or Created the file` you should write `Add style or Create the file`
- ***I write it as optional. But this is a [general requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md#:~:text=The%20commit%20message%20should%20be%20written%20in%20the%20imperative%20tense%20and%20capitalized) for every project. So you should pay attention to this in your next projects.***

**OPTIONAL**

- You can change them by using `git-rebase`. If you want to know how to change ı leave a few resources.
- [Video - How to change commit messages](https://www.loom.com/share/00771f369b7b47fc968c019ecf0d3ba6)
- [Article - How to create good commit messages](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8/)
___

### Cheers and Happy coding! 💻 👏
almost

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Comment by estebanmual** [comment]
image

**Review by Lordkaito — APPROVED** [review]
Hi @estebanmual, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**